WhateverGreen/kern_igfx.hpp

```diff
@@ -499,6 +499,93 @@ class IGFX {
 		static bool forceWakeWaitAckFallback(uint32_t, uint32_t, uint32_t);
 		static void forceWake(void*, uint8_t set, uint32_t dom, uint32_t);
 	} ForceWakeWorkaround;
+	
+	/**
+	 *  Interface of a submodule to fix Intel graphics drivers
+	 */
+	class PatchSubmodule {
```

Comment:
I'm unsure about this polymorphic design. It would be beneficial to store generalised patch pointers in a vector and iterate over those, but we are not doing that, so the advantage is lost.

Reply:
That's my proposal and expectation mentioned above. It would be easier to make new patches if we separate the giant IGFX module into multiple tiny submodules. Yes, ideally IGFX should store a collection of pointers to submodules and iterate over them.